tests/testthat/test-spaces_left_parentheses_linter.R

```diff
@@ -77,3 +77,11 @@ test_that("returns the correct linting", {
 
   expect_lint("a <- -(b)", NULL, spaces_left_parentheses_linter)
 })
+
+test_that("doesn't produce a warning", {
+  expect_silent(
+    lint(
+      system.file("example/complexity.R", package = "lintr")
```

Comment:
the test as written is not very clear, I can imagine trying to revisit this test on 2 years and have no idea what it's doing.

could you elaborate the comment and/or extract a more essential example from that script for clarity?

Reply:
I'll try digging deeper for the cause.

Reply:
![tmp](https://user-images.githubusercontent.com/2363178/101223628-bbdbd600-368c-11eb-9c8b-c056b9a50af4.PNG)

It appears to happen in nested `if` statements if there is a unary minus in the conditional of an inner `if`.
This means a repro is basically the same as the `complexity.R` function.

(Look at the line numbers from parsed content identifying siblings)

I'll add a comment what part of `complexity.R` triggers the bug.

Reply:
great, works for me